https://www.sciencedirect.com/science/article/pii/S0364591616300207?via%3Dihub#s0075
--- src/main.F.orig	2022-03-19 12:25:27.968614000 +0000
+++ src/main.F	2022-03-19 12:26:34.804587382 +0000
@@ -3517,7 +3517,7 @@
 !   gives a good trial step
 !-----------------------------------------------------------------------
         FACT=0
-        IF (DYN%ISIF<5) FACT=10*DYN%POTIM*EVTOJ/AMTOKG *1E-10_q
+        IF (DYN%ISIF<5 .OR. DYN%ISIF==8) FACT=10*DYN%POTIM*EVTOJ/AMTOKG *1E-10_q
         LSTOP2=.TRUE.
 
         NI=1
@@ -3547,7 +3547,7 @@
            ENDDO
         ENDIF
 ! For DYN%ISIF =7 take only pressure (volume relaxation)
-        IF (DYN%ISIF==7) THEN
+        IF (DYN%ISIF==7 .OR. DYN%ISIF==8) THEN
            DO I=1,3
            DO J=1,3
               D2SIF(J,I)=0
